scripts/coverage/python_coverage.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# This script helps to generate coverage report (.xml) for python code(s3backgrounddelete
+# and s3recovery).
+
+set -e
+
+#variables
+SCRIPT_PATH=$(readlink -f "$0")
+SCRIPT_DIR=$(dirname "$SCRIPT_PATH")
+SRC_DIR="$(dirname "$(dirname "$SCRIPT_DIR" )")"
+
+DES_DIR=${DES_DIR:-"/root/code_coverage"}
+
+die() {
+  echo "${*}"
+  exit 1
+}
+
+check_prerequisite() {
+    # install coverage.py package if not present
+    if ! command -v coverage &>/dev/null; then
+        pip3 install coverage
+    fi
+}
+
+run_ut() {
+    echo "Executing UTs"
+    $SRC_DIR/s3backgrounddelete/scripts/run_all_ut.sh
+    $SRC_DIR/s3recovery/scripts/run_all_ut.sh
+}
+
+run_coverage() {
+    check_prerequisite
+    
+    pushd "$SRC_DIR/s3recovery"
+    # Running coverage.py tool over UTs
+    coverage run -m pytest $SRC_DIR/s3backgrounddelete/ut/cortx_s3_* $SRC_DIR/s3recovery/ut/s3recovery_* $SRC_DIR/s3backgrounddelete/s3backgrounddelete/*.py $SRC_DIR/s3recovery/s3recovery/*.py
+
+    coverage report -m $SRC_DIR/s3backgrounddelete/ut/cortx_s3_* $SRC_DIR/s3recovery/ut/s3recovery_* $SRC_DIR/s3backgrounddelete/s3backgrounddelete/*.py $SRC_DIR/s3recovery/s3recovery/*.py
+    # Generate xml report
+    coverage xml -o "$DES_DIR/s3server_python_coverage.xml"
+    popd
+}
+
+check_report() {
+    test -f "$DES_DIR/s3server_python_coverage.xml" || die "Failed to generate the Coverage report."
+}
+
+run_ut
```

Comment:
make an explicit call to check perquisites()
Also write a comment as this is 'main' start of script

Reply:
all 4 scripts

Reply:
Done